modin/experimental/core/execution/native/implementations/hdk_on_native/calcite_serializer.py

```diff
@@ -256,7 +256,8 @@ def serialize_literal(self, literal):
         dict
             Serialized literal.
         """
-        if literal.val is None:
+        val = literal.val
+        if val is None or isinstance(val, (float, np.float64)) and np.isnan(val):
             return {
                 "literal": None,
                 "type": "BIGINT",
```

Comment:
Why is `BIGINT` used for float numbers?

Reply:
I tried DOUBLE, but it fails.
@ienkovich what's the proper way to serialize nan?

Reply:
When we pass it to HDK,  `target_type` can be used to specify a type for the `NULL` literal. But in HDK we need all 7 fields of a literal to present for correct parsing, so this particular path probably works only because we process JSON in Calcite which can add some additional casts, etc. As a general rule:
`type`, `scale`, `precision` - define a type of a value stored in the `literal` field. Any type can be used if `literal` is NULL.
`target_type`, `type_scale`, `type_precision` - define a type of the constant expression we want to have in HDK IR

Reply:
Thanks to @ienkovich, it has been found to work correctly with the following values:
```python
                    "type": "DOUBLE",
                    "target_type": "DOUBLE",
```